jq_dict_parser.go

```diff
@@ -0,0 +1,45 @@
+package opslevel_jq_parser
+
+import (
+	"fmt"
+
+	libjq_go "github.com/flant/libjq-go"
+	"github.com/rs/zerolog/log"
+)
+
+type JQDictParser map[string]JQFieldParser
+
+func NewJQDictParser(dict map[string]string) map[string]JQFieldParser {
+	output := make(map[string]JQFieldParser)
+	if dict == nil {
+		return output
+	}
+	for key, expression := range dict {
+		prg, err := libjq_go.Jq().Program(expression).Precompile()
```

Comment:
It seems like we would want to lift `libjq_go.Jq()` out of this for loop and use the returned `Jq` pointer here.
Inside `Jq()` is `jq.NewJq()`:
```go
func NewJq() *Jq {
	return &Jq{}
}
```
that returns a pointer to a new `Jq` struct every time its called.

Reply:
@davidbloss I added your change, I think this would be better since it looks like the `Jq` struct contains a cache (not sure how it's implemented, but it could be useful). [Source](https://pkg.go.dev/github.com/flant/libjq-go@v1.6.2/pkg/jq#Jq)